tools/codegen/gen.py

```diff
@@ -312,31 +312,65 @@ def go(f: NativeFunction) -> Optional[str]:
             return None
 
         name = cpp.name(f.func)
-
         cpp_returns_type = cpp.returns_type(f.func.returns)
-        cpp_args = cpp.arguments(f.func)
-        cpp_args_str = ', '.join(map(str, cpp_args))
+        cpp_binding = cpp.binding(f.func)
 
         if target is Target.DECLARATION:
-            return f"CAFFE2_API {cpp_returns_type} {name}({cpp_args_str});"
+            if cpp_binding.gathered_signature is None:
+                # There's no TensorOptions
+                result = f"CAFFE2_API {cpp_returns_type} {name}({cpp_binding.signature.cpp_arguments_str()});"
+            else:
+                result = f"CAFFE2_API {cpp_returns_type} {name}({cpp_binding.gathered_signature.cpp_arguments_str()});\n"
+                result += f"CAFFE2_API {cpp_returns_type} {name}({cpp_binding.signature.no_defaults().cpp_arguments_str()});"
```

Comment:
It's almost definitely worth a comment explaining why without a gathered signature you don't need to strip defaults, but with a gathered signature you DO need to strip defaults.